Variables and forms for setting standalone battery timestep

brtietz opened the pull request:
Add new page for stand alone battery timestep, and add it to the "battery" group before the cell and system info.

List numbers don't seem to translate directly into ssc, so add some lk to deal with this.

Mismatches with load, revenue, and curtailment lengths are enforced within ssc as appropriate, the other battery models don't have red text or pop up warnings for these, so I haven't added them here.

Does this require changes to the version upgrade script, or is that a separate effort? 

Goes with https://github.com/NREL/ssc/pull/1227

Commit:
Variables and forms for setting standalone battery timestep

mjprilliman commented:
It looks like the 'Battery Standalone Timestep.json' UI form didn't make it into the commit here.

Commit:
Add new UI form

mjprilliman reviewed (commented):
This error message could use some additional text as there is no weather file (BTM standalone battery mismatch between timestep and load timestep). 

![image](https://github.com/user-attachments/assets/aecebc60-49e9-4774-ad65-e3815dd81e8a)

On the UI form, fit the group box to the contents, give a larger buffer between the text and the group box for different fonts on different platforms, and reduce the width of the dropdown some. 

On version upgrade, maybe set timestep_minutes and timestep_minutes_ui to the corresponding value from curtailment, revenue, or load length from the old case?

Commit:
Add standalone battery variables to fuel cell UI (now required for prechecks)

Commit:
PR request UI improvements, version upgrade script

mjprilliman reviewed (approved):
Looks good, just a couple nitpicks and some things to check when going through the release checklist.

Commit:
Update variable label so version update is less confusing

cpaulgilman commented on deploy/runtime/ui/Battery Standalone Timestep.json:
```diff
@@ -0,0 +1,184 @@
+{
```
I redid this a bit, so it should work across platforms:

![image](https://github.com/user-attachments/assets/408a6941-5c76-40b8-bf4f-f9245932e214)

cpaulgilman commented:
@brtietz, @mjprilliman Any objections to moving this form to the Battery Cell and System page so that it appears under the "Chemistry" group? Right now when you create a new Standalone Battery case, all you see is the simulation time step input with the Battery item in the navigation collapses, which might cause some folks to panic.

cpaulgilman committed:
Refine standalone battery time step form

brtietz commented:
> @brtietz, @mjprilliman Any objections to moving this form to the Battery Cell and System page so that it appears under the "Chemistry" group? Right now when you create a new Standalone Battery case, all you see is the simulation time step input with the Battery item in the navigation collapses, which might cause some folks to panic.

No objections to moving it. I trust your intuition on user expectations & panic levels.

cpaulgilman committed:
Fix white space above Power Converters on Battery Cell and System page

cpaulgilman committed:
Move battery time step

This makes Battery Cell and System the page that loads for standalone battery configurations instead of the Battery Time Step page

cpaulgilman commented:
> > @brtietz, @mjprilliman Any objections to moving this form to the Battery Cell and System page so that it appears under the "Chemistry" group? Right now when you create a new Standalone Battery case, all you see is the simulation time step input with the Battery item in the navigation collapsed, which might cause some folks to panic.
> 
> No objections to moving it. I trust your intuition on user expectations & panic levels.

I moved it below Battery Cell and System as a compromise -- simpler implementation than putting it on Battery Cell and System page and avoids the potential panic.

![image](https://github.com/user-attachments/assets/5ea5e9f8-0b10-42e2-b543-ebea84e2835c)

cpaulgilman committed:
Help ID for new Battery Time Step topic
